modules/cli/src/main/scala/scala/cli/commands/SetupIde.scala

```diff
@@ -40,63 +41,103 @@ object SetupIde extends ScalaCommand[SetupIdeOptions] {
-      downloadDeps(inputs, options.buildOptions, logger).orExit(logger)
 
-    val argv = {
-      val commandIndex = rawArgv.indexOf("setup-ide")
-      val withBspCommand =
-        if (commandIndex < 0) rawArgv // shouldn't happen
-        else rawArgv.take(commandIndex) ++ Array("bsp") ++ rawArgv.drop(commandIndex + 1)
-
-      // Ensure the path to the CLI is absolute
-      val progName = rawArgv(0)
-      if (progName.contains(File.pathSeparator)) {
-        val absoluteProgPath = os.FilePath(progName).resolveFrom(Os.pwd).toString
-        absoluteProgPath +: withBspCommand.drop(1)
+    def downloadDepsIfNeeded(): Either[BuildException, Unit] =
+      options.buildOptions.classPathOptions.extraDependencies.length match {
+        case 0 => Right()
+        case _ => downloadDeps(inputs, options.buildOptions, logger).map(_ => ())
       }
-      else withBspCommand
-    }
-
-    val name = options.bspName.map(_.trim).filter(_.nonEmpty).getOrElse("scala-cli")
-
-    val details = new BspConnectionDetails(
-      name,
-      argv.toList.asJava,
-      Constants.version,
-      scala.build.blooprifle.internal.Constants.bspVersion,
-      List("scala", "java").asJava
-    )
-
-    val gson = new GsonBuilder().setPrettyPrinting().create()
 
-    val json = gson.toJson(details)
+    downloadDepsIfNeeded().flatMap(_ => rawArgs).map { rawArgv =>
+      val dir = options.bspDirectory
+        .filter(_.nonEmpty)
+        .map(os.Path(_, Os.pwd))
+        .getOrElse(inputs.workspace / ".bsp")
 
-    val dir = options.bspDirectory
-      .filter(_.nonEmpty)
-      .map(os.Path(_, Os.pwd))
-      .getOrElse(inputs.workspace / ".bsp")
+      val bspName            = options.bspName.map(_.trim).filter(_.nonEmpty).getOrElse("scala-cli")
+      val bspJsonDestination = dir / s"$bspName.json"
+      val scalaCliBspJsonDestination = inputs.workspace / ".scala" / "ide-options.json"
 
-    val dest = dir / s"$name.json"
-
-    val charset = options.charset
-      .map(_.trim)
-      .filter(_.nonEmpty)
-      .map(Charset.forName)
-      .getOrElse(Charset.defaultCharset()) // Should it be UTF-8?
+      // Ensure the path to the CLI is absolute
+      val absolutePathToScalaCli: String = {
+        if (rawArgv(0).contains(File.separator))
+          os.FilePath(rawArgv(0)).resolveFrom(Os.pwd).toString
+        else {
+          val absoluteScalaCliPath =
+            this.getClass.getProtectionDomain.getCodeSource.getLocation.toURI.getPath.trim
+          if (absoluteScalaCliPath.isEmpty) rawArgv(0) else absoluteScalaCliPath
+        }
+      }
 
-    os.write.over(dest, json.getBytes(charset), createFolders = true)
+      val unparsedArgs =
+        if (args.unparsed.isEmpty)
+          args.unparsed
+        else "--" +: args.unparsed
+      val remainingArgs = args.remaining.map { arg =>
+        if (arg.contains(File.separator))
+          os.FilePath(arg).resolveFrom(Os.pwd).toString
+        else arg
```

Comment:
It seems this won't handle file arguments like in `scala-cli setup-ide Foo.scala` or `scala-cli setup-ide ..`. I think we should resolve the path in all cases, like
```suggestion
        os.Path(arg, Os.pwd).toString
```

(This won't handle URIs or `-` - for stdin - fine, but we can deal with those later, by rejecting them I guess.)